### ISSUE TITLE
Break out more industries in "Industry Category" subscript

### ISSUE THREAD

**Issue by jrissman** [Energy-Innovation/eps-us#89]
Breaking out more industries from the "other industries" group (of from other aggregations we have today, like NGPS, which includes "coke and refined products") could improve accuracy of the IO model's calculated economic impacts.  This is true even if the user applies the same policy settings to each broken-out industry, because it allows them each to use their own job intensity data and other such data.

It might be easiest to add industry categories that (1) have broken-out energy use data in EIA AEO for the U.S. and (2) have a corresponding, matching ISIC code in OECD data.  This might also tend to break out many of the most important industries.

On 8/26/20, Robbie made the following table indicating which OECD ISIC codes have matching, broken-out industries in AEO data.  A value of 0.5 indicates there are either multiple ISIC codes or multiple AEO categories.

ISIC Code | ISIC Code meaning | AEO match?
-- | -- | --
ISIC 01T03 | Agriculture, forestry and   fishing | 1
ISIC 05T06 | Mining and extraction of energy producing products | 0.5
ISIC 07T08 | Mining and quarrying of non-energy producing products | 0.5
ISIC 09 | Mining support service activities | 0.5
ISIC 10T12 | Food products, beverages and tobacco | 1
ISIC 13T15 | Textiles, wearing apparel, leather and related products | 1
ISIC 16 | Wood and of products of wood and cork (except furniture) | 1
ISIC 17T18 | Paper products and printing | 1
ISIC 19 | Coke and refined petroleum products | 1
ISIC 20T21 | Chemicals and pharmaceutical products | 1
ISIC 22 | Rubber and plastics products | 1
ISIC 23 | Other non-metallic mineral products | 1
ISIC 24 | Manufacture of basic metals | 1
ISIC 25 | Fabricated metal products, except machinery and equipment | 1
ISIC 26 | Computer, electronic and optical products | 1
ISIC 27 | Electrical equipment | 1
ISIC 28 | Machinery and equipment n.e.c. | 1
ISIC 29 | Motor vehicles, trailers and semi-trailers | 1
ISIC 30 | Other transport equipment | 0.5
ISIC 31T33 | Other manufacturing; repair and installation of machinery and equipment | 1
ISIC 35T39 | Electricity, gas, water supply, sewerage, waste and remediation services | 0
ISIC 41T43 | Construction | 1
ISIC 45T47 | Wholesale and retail trade; repair of motor vehicles | 0
ISIC 49T53 | Transportation and storage | 1
ISIC 55T56 | Accomodation and food services | 0
ISIC 58T60 | Publishing, audiovisual and broadcasting activities | 0
ISIC 61 | Telecommunications | 0
ISIC 62T63 | IT and other information services | 0
ISIC 64T66 | Financial and insurance activities | 0
ISIC 68 | Real estate activities | 0
ISIC 69T82 | Other business sector services | 0
ISIC 84 | Public administration and defence; compulsory social security | 0
ISIC 85 | Education | 0
ISIC 86T88 | Human health and social work | 0
ISIC 90T96 | Arts, entertainment, recreation and other service activities | 0
ISIC 97T98 | Private households with employed persons | 0

_Suggested by Robbie on 8/26/20_

**Comment by jrissman** [comment]
Per our discussion today, Olivia suggests we split up the NGPS industry into three industries: extraction, refining, and distribution.  This requires splitting the methane leakage in BPEiC and methane leakage abatement potential in PERAC between the extraction and the distribution industries.

We are still not splitting up oil and gas (which are often co-produced).

Gas processing (separating different hydrocarbons from methane in gas streams, often done at a separate facility not terribly far from the point of extraction) will be included within the "extraction" industry, as it is too minor to qualify as its own industry, and grouping it with refining doesn't make sense.

EIA AEO has data on natural gas distribution energy use in the transportation sector (pipeline fuel), but we will be using that data in the industry sector in the EPS.

**Comment by jrissman** [comment]
In response to issue #130, ISIC 05T06 was split into coal mining (ISIC 05) and oil and gas extraction (ISIC 06).

[This site](https://siccode.com/isic-code-lookup-directory) has a searchable, browsable list of ISIC codes down to the four-digit level.  [This UN document](https://unstats.un.org/unsd/publication/seriesm/seriesm_4rev4e.pdf) has the ISIC codes in PDF format.

Our expansion of the `Industry Category` subscript will be based on which industries' data are available broken-out in the EIA AEO.  For the industry sector, here are the available industry categories in AEO, and their matching ISIC codes:

* Refining - ISIC 19 (technically ISIC 192, as 191 is coke)
  * Note that AEO's table breaks out ethanol plants, which supply biofuels, which may not be part of ISIC 19 as they are not petroleum products.  Biofuels aren't mentioned in any ISIC code.   Biofuels are likely part of chemicals (ISIC 20).
* Food - ISIC 10T12 (assuming the AEO's "food" data includes beverage and tobacco manufacturing)
* Paper - ISIC 17 (in EPS, ISIC 17 is grouped with ISIC 18, printing and reproduction of recorded media)
* Bulk chemicals - ISIC 20
* Glass - ISIC 231 (part of ISIC 23, non-metallic minerals)
* Cement and lime - ISIC 2394, part of ISIC 23, non-metallic minerals)
* Iron and steel - ISIC 241 (part of ISIC 24, manufacture of basic metals)
* Aluminum - ISIC 242 (part of ISIC 24, manufacture of basic metals)
* Metal-Based Durables (table in AEO containing multiple broken-out industries)
  * Fabricated Metal Products (excludes machinery and vehicles) - ISIC 25
  * Machinery - ISIC 28
  * Computers - ISIC 26
  * Transportation Equipment - ISIC 29T30 (this is broken into ISIC 29 and ISIC 30 in the EPS, need to apportion AEO data)
  * Electrical Equipment - ISIC 27
* Other manufacturing sector energy consumption (table in AEO containing multiple broken-out industries)
  * Wood products - ISIC 16
  * Plastics - ISIC 2013 (part of ISIC 20, chemicals)
  * Balance of manufacturing (apportion to manufacturing ISIC codes not used above - e.g. 13-15, 21, 31-32)
* Nonmanufacturing sector energy consumption (table in AEO containing multiple broken-out industries)
  * Agriculture - ISIC 01 (and 03 for aquaculture) - part of ISIC 01T03
  * Construction - 
  * Mining - ISIC 05T05 (this is broken into ISIC 05, ISIC 06, and ISIC 07T08 in the EPS, and we can preserve the existing methodology in `indst/BIFUbC` for breaking down the "mining" energy use into separate industry categories.  We likely want to add an industry category for mining of non-energy-producing products that maps onto ISIC 07T08.  Consider whether to group uranium mining with coal mining or with 07T08.)

* We also have one existing industry category, water and waste (previously referred to as "waste management," that doesn't map onto AEO data.  We will be preserving this category and its non-AEO data source.

Therefore, it looks like the updated Industry Category subscript will have the following elements (not in this order):

1. refining of petroleum (uses only part of the data from 1 AEO table - excludes biofuels)
2. food, beverage, and tobacco (1-to-1 match with AEO data table)
3. pulp, paper, and printing (1-to-1 match with AEO data table)
4. chemicals, plastics, and biofuels (groups data from 3 AEO tables)
5. glass, cement, concrete, lime, brick, and tile (groups data from 2 AEO tables - glass and cement)
6. iron and steel (1-to-1 match with AEO data table)
7. aluminum and other metals (1-to-1 match with AEO data table, because this category will only contain aluminum for the U.S. EPS)
8. fabricated metal products excluding machinery and vehicles (1-to-1 match with AEO data table)
9. machinery (1-to-1 match with AEO data table)
10. computers (1-to-1 match with AEO data table)
11. electrical equipment (1-to-1 match with AEO data table)
12. road vehicles (LDVs, HDVs, motorbikes) (need to split up the data in AEO "Transportation Equipment" table)
13. non-road vehicles (aircraft, rail, ships) (need to split up the data in AEO "Transportation Equipment" table)
14. wood products (1-to-1 match with AEO data table)
15. agriculture (1-to-1 match with AEO data table)
16. construction (1-to-1 match with AEO data table)
17. oil and gas extraction (need to split up the data in AEO "Mining" table)
18. coal mining (need to split up the data in AEO "Mining" table)
19. other industries (1-to-1 match with AEO data table)

* Optionally, I may consider splitting ISIC 23 in the EPS into ISIC 231 (glass) and ISIC 239 (other non-metallic minerals), if I wish to handle the AEO data for glass and for cement separately.

**Comment by jrissman** [comment]
Attached is an Excel file that details the final set of Industry Categories and ISIC Codes for 3.2.0, along with which AEO table(s) contain data for each one.  Industry Categories will be a strict subset of ISIC codes, which will make mapping easier.  The attached spreadsheet is based on a lot of careful thought and weighing many considerations.

### [2021-01-14 New Industries and ISIC Codes.xlsx](https://github.com/Energy-Innovation/eps-us/files/5818452/2021-01-14.New.Industries.and.ISIC.Codes.xlsx)

Each actual Industry Category element name in the model will likely include the appended number of its corresponding ISIC code, to help increase clarity and reduce the chances of error.

**Comment by robbieorvis** [comment]
This is awesome; we are going to have so much detail now!


________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D6EB23.6D8DF190]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
Note that there is a slight mismatch between U.S. economic data availability and U.S. energy use data availability.  We have economic data for all ISIC codes (including all industry categories) from BLS and BEA, fully disaggregated.  However, when it comes to energy use, the AEO is quite good, but it doesn't quite provide complete energy use data for every industry category.  The discrepancies are:
* food beverage and tobacco (AEO only breaks out food, not beverage or tobacco)
* textiles apparel and leather (AEO doesn't break out any of these)
* pulp paper and printing (AEO only breaks out pulp and paper, not printing)
* other metals (AEO only breaks out aluminum)

Anything AEO doesn't "break out" is included in AEO's "balance of manufacturing" category.

(AEO groups "furniture" with "balance of manufacturing" rather than with "wood products," but this is also how the economic data is categorized, so that one is not a discrepancy.)

These discrepancies are minor.  Nonetheless, to maximize accuracy and completeness, I plan to re-assign some of the energy use in AEO's "balance of manufacturing" category to these other four categories, likely based on output ratios from the BEA data.

When adapting the EPS to a region without data broken out for all of the EPS industry categories, **here is what to do**:

1. If the industry doesn't meaningfully exist in that region - say, steel production in Monaco - then just leave its energy use and economic data both set at zero.  Note that this means that in DLIM and SIOM, no industry should draw from the non-existent industry as a supplier, unless the domestic content share of that industry is set to zero (e.g. Monaco can have steel users if they import all their steel).

2. If the industry does exist in that region, but your energy use data is lumped into some other category like "other industries," the main thing to do is to **make sure you use the same grouping for your economic data as you use for your industry data**.  So if you have jobs, output, value added, EE compensation, and so forth assigned to "iron and steel," but iron and steel's energy use is part of "other industries," this won't produce the right results.  (It will understate the jobs, GDP, and other economic impacts of energy policies applied to the "other industries" category.)  You'd need to either:
    * break the energy use of "iron and steel" out from "other industries" and assign it to "iron and steel," or
    * reallocate the jobs, output, value added, EE compensation, etc. from "iron and steel" to "other industries."  If you choose to reassign the iron and steel economic data to the other industries category, that includes changing the DLIM and SIOM tables so that suppliers that were buying from the iron and steel industry instead buy the same amount of goods from the "other industries" category.

The use of the "other industries" category above is just an example.  You would use the same procedure if, for example, you were breaking up a "metals" industry into "iron and steel" and "other metals," or grouping "iron and steel" with "other metals" into a single "metals" industry.

In other words, the EPS is going to be set up to be highly flexible, so that you don't need to scramble to collect fully disaggregated data for every one of the 25 industry categories in every EPS region.  You can omit industries that don't exist, and you can group some industries that do exist (but for which you don't have fully broken-out data).  You also have flexibility in how you want to define the new groupings.  This should maximize your ability to adapt the EPS to new regions within the constraints of locally-available data.  The only thing that the EPS cannot tolerate while preserving full accuracy is using different groupings in different variables (e.g. breaking out "iron and steel" in the economic variables but not in the energy use variable, or vice versa).

**Comment by jrissman** [comment]
Commits eb83f1f, 3ac4541, a1f1c3f, 9bf074e, and 01eb738 together fully implement the new ISIC code splits, which should give us what I hope is our final set of ISIC codes.

In this version, there is a problem where the "water and waste" (also known as "waste management") industry, with ISIC code 36T39, grows out-of-hand due to exponential reinforcement through the macroeconomic energy feedback loop.  I took quite a while today trying to isolate and understand this problem.  It turns out to be an issue with the variable `Percent Change in Contribution to Gross Domestic Product Excluding Direct Impacts by ISIC Code`, which calculates the change caused by the policy packages relative to the BAU contribution of each ISIC code to total GDP in percentage terms.  This can affect any industry, but it is particularly problematic for the water and waste industry, because its BAU contribution to GDP is very small.  It is the smallest contribution of all industries.  This means that even relatively modest changes in spending on this industry caused by policy packages can result in very large percentage increases.  This gets further amplified through the feedback loop, resulting in unreasonable results.

I've never been 100% happy with the way the macroeconomic energy use feedback loops are calculated, because they are so sensitive to particularities of the input data, to avoid exponential reinforcement or oscillation.  While we got it working well for the U.S. (prior to the latest ISIC code split), it was always a risk when adapting the model to other regions.  And now, in the national U.S. version, I did not find any problems with the input data for the "water and waste" industry.  I think it legitimately contributes little to GDP (small Value Added), and it legitimately gets a boost from policies that increase spending.  The same thing could be true of some other industry in a U.S. state or foreign country.

I think we need to change the way this feedback loop is calculated to be based on addition and subtraction, rather than multiplication and division (e.g. in `Percent Change in Contribution to Gross Domestic Product Excluding Direct Impacts by ISIC Code`).  This would be similar to the way we fixed issue #131, where noise was being amplified due to the way we calculated something with division.  In this case, it isn't noise, but it is a small signal that is being way over-amplified.

This is a good example of how one gradually learns and refines best modeling practices.  It isn't always enough that an equation is mathematically correct in an abstract sense - it needs to be robust to different input conditions.  For example, it turns out that there are certain places where it is perilous to use division to calculate something - particularly when the denominator might become extremely small, and particularly when that calculation is part of a feedback loop.

The best way to fix it is not immediately obvious, because the GDP changes are in units of dollars, but the feedback loops produce changes in vehicle-distance traveled, energy-using building services, and industrial production, none of which are calculated in dollars.  We currently use a percent change on the GDP to make it unitless, so we can apply it cleanly across multiple sectors.   I'll need to work something out next week.

Note that if you've been waiting for a final set of files with updated ISIC codes to share with external partners who are developing procedures to populate these files, I think the set in the latest commit in branch "#89" should already be final.

**Comment by jrissman** [comment]
14e51da fixes the issue with the "water and waste" sector.  It turned out not to be a problem with how `Percent Change in Contribution to Gross Domestic Product Excluding Direct Impacts by ISIC Code` is calculated.  It was a mis-alignment between how this industry's revenues were provided in the industry sector (in `indst/BNRbI`) and in the IO calculations (in `io-model/BObIC`).  Revenue was over 10x higher in the industry sector, which led to unrealistically large changes in revenue due to production changes, particularly when taken as a percentage of output from `BObIC`.  We don't actually need `BNRbI` anymore - it is a hold-over from before we had the IO model, and now that we have fully disaggregated the ISIC codes, we can map specific ISIC codes in `BObIC` onto the industry categories.  So I removed `BNRbI` and use `BObIC` instead, and this ensures alignment between the revenue figures used in the Industry sector and in the IO model, and fixes the issue.

As a separate matter, while investigating this, I realized that the following assertion in the `Macroeconomic Feedbacks` sheet in the `Energy Service Demand Feedbacks` section may not be right in all cases: _"In the demand sectors' energy demand feedback loops, we want to feed back in only the percent change in demand caused by the indirect and induced effects of the policy package (as the direct effects are already included in each of these sectors)."_  The energy demand feedback loops are used in three sectors: Transportation, Buildings, and Industry.  I was thinking of (for example) transport, where direct increases or decreases in vehicle-distance traveled are already reflected, and you don't want to double-count by having the direct effects of the IO model also affect travel demand.  But direct effects from another sector are not necessarily reflected in the Industry sector.  For instance, the 1.5-degree scenario reduces the revenue of the "water and waste" industry in the Electricity sector, probably because it is switching to power plant types with lower water demand.  That is a direct reduction due to the policy package, so it isn't included in the feedback loop.  The indirect and induced effects on that industry increase revenue.  (Some of government and household increased spending goes there, and some of the businesses' increased economic activity is spent on water and waste services.)  So the indirect and induced effects on "water and waste" are positive.   The positive indirect and induced effects are about +4 billion, while the direct reduction in "water and waste" spending is about -8 billion.  So in other words, this industry is slightly smaller overall in the policy case.  However, by excluding direct effects from the energy service demand feedback loop, it only sees the +4 billion.  So we see a slight growth in the "water and waste" industry, when we should be seeing a slight decline.

I haven't mapped out exactly the boundaries of where we need to include and exclude effects from the energy demand feedback loops.  It's tricky.  We don't currently monitor or track which sector an effect arises from - if we have to start doing that, we'll need a new, small calculation structure to do it.  We also likely have to treat fuel industries different from non-fuel industries, because the fuel industries are handled by fuel demand from all sectors, so they are affected by the feedback loop indirectly (i.e. when it boosts fuel-consuming industries), not directly, as other industries are.

Overall, the effect of the energy demand macroeconomic feedback loop are not that large, and I think we are already including most of the financial impacts we want in there, so I don't expect this issue to be affecting the results too much.  But it is something we should look to resolve through more careful tracking of which effects to include or exclude when calculating the energy demand macroeconomic feedbacks.

**Comment by jrissman** [comment]
I've been working on updating the Industry Categories in process emissions variables, starting with BPEiC.  Some observations in doing this:

- We had previously assigned all F-gas emissions to the chemicals industry, because we assign these emissions to the industry that produces the gases, even if the gases are sold to a different industry or consumer and are emitted later.  This is correct for all but one type of F-gas emissions.  The exception is F-gas emissions from the aluminum industry.  These are produced by the aluminum industry itself as an unwanted byproduct, not purchased from the chemicals industry.  They represented 36 MMT out of a total 1136 MMT in 2018 (that's 3%), in the EPA data.  In the EPS 3.2 update, F-gases from aluminum manufacture will be assigned to the "other metals" industry rather than the chemicals industry.

- I think the purpose of the F-gas "inspect, maintenance, retrofit" lever might have initially been to target these aluminum-related F-gas emissions.  If this policy lever doesn't meaningfully target F-gas emissions coming from the chemicals industry, I will reassign this lever to only affect the "other metals" industry, while the other three F-gas levers (F-gas substitution, destruction, and recovery/recycling) all will continue to affect the chemicals industry.  I can also potentially move some of the mitigation measures in PERAC that are currently encompassed by the "inspect, maintenance, retrofit" lever to one of the other three F-gas levers, if needed, so overall abatement potential will remain unchanged, but which F-gas lever lever controls which potential might shift very slightly.

- Due to the way the old "natural gas and petroleum systems" industry category is being split up into "oil and gas extraction" "refining" and "energy pipelines and gas processing" industry categories in EPS 3.2, methane leakage that used to be assigned all to one industry now needs to be split among three.  This will require splitting up PERAC abatement levers as well.  This is very fiddly, time-consuming work, but I think it is necessary for proper mapping of process emissions onto the new industry categories.  (Even splitting up BPEiC into more industry categories is annoyingly difficult, given the way BPEiC now has so many exceptions built into it.)

If there is appetite and spare time in the EPS team to do some of this Excel-based work, I'm happy to let one of you handle the updates to BPEiC and PERAC for 3.2.  This would enable EPS 3.2 to get done quicker.  If this is of interest, please let me know, and I can send you an email with more details on what needs to be done to BPEiC and PERAC.  If not, that's perfectly okay - I'm happy to do it.  I just wanted to give you the option.

**Comment by jrissman** [comment]
It looks like it may be advantageous for me to complete issue #116 as part of the work on this issue, #89.  Issue #116 is asking for a different implementation of industrial fuel shifting, which provides separate levers for switching to hydrogen and to electricity, instead of one lever that switches to a fuel mix defined in input data.  Having a couple levers with their own target fuel settings removes the need for difficult calculations in input data to find the electrification potential of different industries, which is harder when the industries are more disaggregated.  Therefore, from an input data perspective, it may be advantageous to do #116 as part of #89.

Note that the way I implement #116 would not literally involve building levers that are hard-coded to shift to electricity and hydrogen, because that would remove some valuable capabilities that exist in the current implementation.  For instance, we can currently shift to any fuel mix, so we can handle shifting to biomass, potentially relevant in countries like Brazil that use bio-energy significantly in Industry, as well as handling coal-to-gas shifts, relevant in certain places with localized air quality concerns.  The new implementation would likely involve essentially duplicating the existing fuel shifting lever and associated input data, and then the total shifts would be the sum of the settings of the two levers (capped at 100% total, of course).

**Comment by robbieorvis** [comment]
This sounds good.

One question – are there fuels beyond biomass, hydrogen, and electricity we would want to shift to? If not, it may make sense to just have three of these levers for those fuels.

FWIW, I think part of the request from partners relates specifically to the use of different policy schedules (something to consider as you develop this). For example, we wanted to be able to push hard on electrification now but defer fuel switching to hydrogen until the future.

**Comment by jrissman** [comment]
Already two levers for one policy (Industrial Fuel Shifting) is a lot (we don't have any other policies that use two levers), and is somewhat undesirable.  Three levers would be worse.  This is what subscripting is for.

You mention electricity, biomass, and hydrogen as three target fuels we want to support via user-set lever settings.  You ask whether there are any others.  The one that comes to mind is natural gas.  Coal-to-gas industrial retrofits used to be an important way this lever was used in the U.S. simulator.  Now-a-days, people more often talk about shifting to clean fuels, not using NG as a bridge fuel, but I don't think the concept of coal-to-gas retrofits is dead by any means (it's still pushed by gas utilities and the like), and frankly, it is nice that the EPS currently is able to model this strategy.  I am reluctant to drop support for this.  Someday you might want to model coal-to-gas retrofits, even if it's just to show that some other alternative is better.  Also, coal-to-gas might be of interest in countries where local air quality is more of concern than GHGs.  So that's a fourth target fuel type this lever should be able to handle.  We really need subscripting here, not a plethora of levers.

I think the best approach to manage complexity here is not to eliminate the _target_ fuel subscript, but to eliminate the _source_ fuel subscript.  For a given industry, all fuel types flagged in a new Boolean input data file would shift to the combination of fuels specified by the user via the lever settings.  This seems to me the way to avoid needing to introduce a new, third subscript on the lever.  Our web interface doesn't support three subscript on a policy lever, and it would make the lever a huge pain to set or use, and would make the policy schedule subscript ridiculously long (25 industry categories * 10 source industrial fuels * 10 target industrial fuels = 2,500 levers, accompanied by 2,500 policy schedules).  Even 250 levers is a lot, but that's what we're going to have after the Industry Category expansion even without doing issue #116.  (We don't have to show them all in the web interface.)

Do you anticipate the need, _within a particular industry_, to shift different existing (source) fuels to different combinations of (target) fuels from each other?  For example, suppose in Country X, the Glass Manufacturing industry uses some electricity, some coal, and some natural gas in the BAU case.  A new Boolean input file specifies that coal and natural gas are subject to fuel shifting, while electricity should not be shifted.  Do you need to be able to specify different percentages of coal and of gas to shift (say, shift 30% of coal and 40% of natural gas) to new fuel types, or is it okay if the percentage of Glass Manufacturers' coal and gas use to be shifted is the same (say, 40% of each)?  And do you need to be able to shift them to different fuel types from each other, or is it okay if the coal and gas are shifted to the same (new) fuel mix?  For example, is it okay if Glass Manufacturers' coal use and natural gas use both are shifted to 50% electricity and 50% hydrogen, or do you need the ability to specify that Glass Manufacturers' coal use is shifted 25% to electricity and 75% to hydrogen, while Glass Manufacturers' natural gas use is shifted 50% to electricity and 50% to hydrogen?

I think we probably have to do it that way because we sinply can't introduce a third subscript on the policy lever without unduly increasing program complexity and impacting UI, but I at least want to explain what I think is the best strategy to handle this that's available to us right now.

**Comment by jrissman** [comment]
#89 is completed as of commit d5c4d5e.  I've merged "develop" into #89 to show it works with all the latest data updates.  I haven't merged #89 into "develop" because there may be more "develop" updates going into EPS 3.1.2 before we're ready to release 3.1.2.  I will merge #89 into "develop" after we have a final 3.1.2 for release.

Here's a screenshot of the Industry Fuel Use (excluding feedstocks) graph showing off the new industry categories:

![3 2-IndsFuelUseExclFeedstocks](https://user-images.githubusercontent.com/7120106/108133190-06a0a380-7069-11eb-834b-80f9ad33172e.PNG)

**Comment by jrissman** [comment]
Earlier in this thread, there is some discussion of a problem with the energy feedback loops unreasonably magnifying the size of the "water and waste" industry.  The issue was largely fixed with commit 14e51da on 1/28/21, discussed above.  While that fix solved most of the problem, we were still seeing too much magnification of the water and waste industry in EPS 3.2.0.  After further review today, I discovered an input data error where some revenue for the "water and waste" industry had been accidentally omitted, affecting its output, value added, and EE compensation figures.  I fixed this in commit c93fd26.  Now, the water and waste industry sees percentage growth or shrinkage of similar magnitude as other industries (and as overall GDP) in variable `Percent Change in Contribution to Gross Domestic Product Excluding Direct Impacts by ISIC Code`, so this looks like it's fully fixed.

I did carefully review the structural methodology and could find no flaws there.  I believe the lingering problem (i.e. after 14e51da) was entirely an input data issue.
